tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -357,11 +409,24 @@ class SerializableModelLibraryGenerator {
     ClassDefinition classDefinition,
     List<SerializableModelFieldDefinition> fields,
   ) {
+    var parentClass =
+        classDefinition.extendsClass is ResolvedInheritanceDefinition
+            ? (classDefinition.extendsClass as ResolvedInheritanceDefinition)
+                .classDefinition
+            : null;
```

Comment:
Question: This seems like code duplication of the logic that is part of the entrypoint for the library generated. Should this be placed inside of the classDefinition instead so that it can be reused?

Reply:
Then we could simply ask the classDefiniton for it's parent class.